cmake/BLTPrivateMacros.cmake

```diff
@@ -599,8 +604,12 @@ macro(blt_split_source_list_by_language)
             if (DEFINED arg_Fortran_LIST)
                 list(APPEND ${arg_Fortran_LIST} ${_file})
             endif()
+        elseif(${_ext_lower} IN_LIST BLT_CMAKE_FILE_EXTS OR ${_name} STREQUAL "CMakeLists.txt")
```

Comment:
I elected to treat `CMakeLists.txt` here rather than place it in the list of extensions in `BLT_CMAKE_FILE_EXTS` because an exact filename match seemed like the simplest solution, and the extension-matching code seemed like a poor fit because it's case-insensitive.

Reply:
Ah yes thanks for catching that @kennyweiss .  CMake parses strings first before evaluating if's.  We continually are bit by this.

If not quoted, empty strings do this and produce a very very hard to understand parsing error. The following:

```
if (${foo} STREQUAL "foo")
```

is evaluated as:

```
if ( STREQUAL "foo")
```